File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.py
def solution(n, arr1, arr2):
    answer = []
    map1=[]
    map2=[]

    for i in range(0,n,1) :
        b=format(arr1[i],'b')
        c=format(arr2[i],'b')
        map1.append(('0'*n+b) [-n:])
        map2.append(('0'*n+c) [-n:])
    
    for i in range(n) :
        sharp=''
        for j in range(n) :
            if str(map1[i])[j] == '0' and str(map2[i])[j] == '0' :
                sharp+=' '
            else :
                sharp+='#'
        answer.append(sharp)
                
    print(map1)
    print(map2)
    
    return answer

File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.java
class Solution {
    public String[] solution(int n, int[] arr1, int[] arr2) {
        String[] answer = new String[n];
        
        for(int i=0; i<n; i++){
            String bi_arr1 = Integer.toBinaryString(arr1[i]); 
            while (bi_arr1.length() < n){
                bi_arr1 = "0" + bi_arr1 ;
            }
            
            String bi_arr2 = Integer.toBinaryString(arr2[i]); 
            while (bi_arr2.length() < n){
                bi_arr2 = "0" + bi_arr2 ;
            }
            
            String answer_s = "";
            for(int j=0; j<bi_arr2.length(); j++){
                if(bi_arr1.charAt(j)=='1' || bi_arr2.charAt(j) =='1'){
                    answer_s += "#";
                }else{
                    answer_s += " ";
                }
            }
            answer[i] = answer_s;
        }
        
        
        return answer;
    }
}